src/overlays/actors/ovl_En_Bigslime/z_en_bigslime.c

```diff
@@ -54,289 +243,2910 @@ const ActorInit En_Bigslime_InitVars = {
+    s32 j;
+
+    if (this->freezeTimer) {
+        this->freezeTimer--;
+    }
+
+    this->skelAnime.playSpeed = this->freezeTimer / 40.0f;
+    vtxIceSeed = this->freezeTimer * 4;
+    for (vtxIceUpdate = 0; vtxIceUpdate < 4; vtxIceUpdate++, vtxIceSeed++) {
+        if (vtxIceSeed < BIGSLIME_NUM_VTX) {
+            dynamicVtx = &sBigslimeDynamicVtx[this->dynamicVtxState][vtxIceSeed];
+            targetVtx = &sBigslimeTargetVtx[vtxIceSeed];
+            randFloat = randPlusMinusPoint5Scaled(40.0f);
+            dynamicVtx->n.ob[0] += (s16)(randFloat / this->actor.scale.x);
+            dynamicVtx->n.ob[1] += (s16)(randFloat / this->actor.scale.y);
+            dynamicVtx->n.ob[2] += (s16)(randFloat / this->actor.scale.z);
+            if (((dynamicVtx->n.ob[1] * this->actor.scale.y) + this->actor.world.pos.y) < GBT_ROOM_5_MIN_Y) {
+                dynamicVtx->n.ob[1] = ((GBT_ROOM_5_MIN_Y - this->actor.world.pos.y) / this->actor.scale.y) - 1.0f;
+            }
+
+            for (j = 0; j < 3; j++) {
+                targetVtx->n.ob[j] = dynamicVtx->n.ob[j];
+            }
+        }
+    }
+
+    for (vtxIceUpdate = 4; vtxIceUpdate < BIGSLIME_NUM_VTX; vtxIceUpdate++) {
+        sBigslimeTargetVtx[vtxIceUpdate - 4].n.a = sBigslimeTargetVtx[vtxIceUpdate].n.a;
+    }
+
+    func_800B9010(&this->actor, NA_SE_EV_ICE_FREEZE - SFX_FLAG);
+    if (this->actor.bgCheckFlags & 2) {
+        if (this->freezeTimer == 0) {
+            EnBigslime_BreakIntoMinislime(this, globalCtx);
+        } else {
+            this->bigslimeCollider[0].base.acFlags |= AC_ON;
+            EnBigslime_AddIceShardEffect(this, globalCtx);
+            EnBigslime_SetupSquishFlat(this);
+        }
+    } else if (this->freezeTimer == 0) {
+        if (!(this->actor.bgCheckFlags & 1)) {
+            EnBigslime_SetupFrozenFall(this);
+        } else {
+            EnBigslime_SetupFrozenGround(this);
+        }
+    }
+}
+
+void EnBigslime_SetupFrozenGround(EnBigslime* this) {
+    this->freezeTimer = 80;
+    this->bigslimeCollider[0].base.acFlags |= AC_ON;
+    this->actionFunc = EnBigslime_FrozenGround;
+}
+
+void EnBigslime_FrozenGround(EnBigslime* this, GlobalContext* globalCtx) {
+    f32 invFreezerTimer;
+    s32 randSign;
+    f32 randFloat;
+
+    this->freezeTimer--;
+    if (this->freezeTimer == 0) {
+        EnBigslime_AddIceShardEffect(this, globalCtx);
+        EnBigslime_SetTargetVtxFromPreFrozen(this);
+    } else if (this->freezeTimer == 40) {
+        Math_Vec3f_Copy(&this->iceShardRefPos, &this->actor.world.pos);
+    } else if ((this->freezeTimer < 20) || ((this->freezeTimer < 40) && ((this->freezeTimer % 2) != 0))) {
+        invFreezerTimer = 1.0f / this->freezeTimer;
+        randFloat = Rand_ZeroFloat(4.0f * invFreezerTimer);
+
+        // clang-format off
+        randSign = Rand_ZeroOne() < 0.5f ? -1 : 1; \
+        this->actor.world.pos.x = randSign * (1.0f + invFreezerTimer + randFloat) + this->iceShardRefPos.x;
+        // clang-format on
```

Comment:
Personally, I would probably put the randFloat inside the `clang format` block even though it is not needed, since it seems logically connected.